executor/new_distsql.go

```diff
@@ -355,21 +358,26 @@ func (ih *indexHandler) fetchHandles(e *IndexLookUpExecutor, result distsql.Sele
 	}
 }
 
-func (ih *indexHandler) close() {
+func (ih *indexWorker) close() {
 	ih.wg.Wait()
 }
 
-// tableHandler is used by IndexLookUpExecutor to maintain table lookup background goroutines.
-type tableHandler struct {
-	taskChan chan *lookupTableTask
-	taskCurr *lookupTableTask
+func (e *IndexLookUpExecutor) waitIndexWorker() {
+	e.indexWorker.close()
+}
+
+// tableWorker is used by IndexLookUpExecutor to maintain table lookup background goroutines.
+type tableWorker struct {
+	taskChan chan<- *lookupTableTask
```

Comment:
I think `resultChan` maybe a better name. `task` is the type of the channel, it doesn't give any information.
the table worker pick a task from `workChan`, after finished, send the task to `resultChan`, this is easier to understand.